shark-graph/src/test/java/shark/LongScatterSetTest.kt

```diff
@@ -0,0 +1,131 @@
+package shark
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Ignore
+import org.junit.Test
+import shark.internal.hppc.HHPC
+import shark.internal.hppc.LongScatterSet
+
+class LongScatterSetTest {
+
+  @Test fun `verify add operation`() {
+    val set = LongScatterSet()
+    verifyAddOperation(set)
+  }
+
+  @Test fun `verify remove operation`() {
+    val set = LongScatterSet()
+    verifyRemoveOperation(set)
+  }
+
+  @Test fun `verify release operation`() {
+    val set = LongScatterSet()
+    set.add(42)
+    set.release()
+
+    assertFalse(42 in set)
+    assertEquals(0, set.size())
+  }
+
+  @Test fun `verify ensureCapacity does not break operations`() {
+    // Ensure capacity before adding/removing elements
+    verifyAddOperation(LongScatterSet().apply { ensureCapacity(10) })
+    verifyRemoveOperation(LongScatterSet().apply { ensureCapacity(10) })
+
+    // Ensure capacity after adding/removing elements
+    val set = LongScatterSet()
+    set.add(42)
+    set.add(10)
+    set.ensureCapacity(100)
+
+    assertTrue(42 in set)
+    assertTrue(10 in set)
+    assertEquals(2, set.size())
+  }
+
+  @Ignore(value = "Takes ~20 seconds on MacBook Pro 2019, disabled so it doesn't waste CI time.")
+  @Test(expected = RuntimeException::class)
```

Comment:
Alternatively we can check current free memory, fill it almost completely with an array and then start filling the set - it will be faster.
But I generally don't like an idea of throwing an OOM on CI in the test..

Reply:
Why do we need to test this? Consider removing.